src/Extension/ResizeBrowserWindowExtension/ResizeBrowserWindowListener.php

```diff
@@ -0,0 +1,60 @@
+<?php
+/**
+ * @author    Craig Gosman <craig@ingenerator.com>
+ * @licence   proprietary
+ */
+
+namespace Ingenerator\BehatSupport\Extension\ResizeBrowserWindowExtension;
+
+use Behat\Behat\EventDispatcher\Event\BeforeScenarioTested;
+use Behat\Behat\EventDispatcher\Event\ScenarioTested;
+use Behat\Mink\Mink;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+class ResizeBrowserWindowListener implements EventSubscriberInterface
+{
+    /**
+     * @var Mink
+     */
+    protected $mink;
+
+    /**
+     * @var array
+     */
+    protected $options;
+
+    /**
+     * @param Mink  $mink
+     * @param array $options
+     */
+    public function __construct(Mink $mink, array $options)
+    {
+        $this->mink    = $mink;
+        $this->options = $options;
+    }
+
+    public static function getSubscribedEvents()
+    {
+        return [
+            ScenarioTested::BEFORE => 'beforeScenario',
+        ];
+    }
+
+    /**
+     * Hook fired before every scenario that starts (or checks status of) phantomjs when first required
+     *
+     * @param BeforeScenarioTested $event
+     */
+    public function beforeScenario(BeforeScenarioTested $event)
+    {
+        if ( ! $event->getScenario()->hasTag('javascript')) {
+            return;
+        }
+
```

Comment:
Probably best to add a state flag here to return if this has already run (so it only actually does anything on the first JS scenario in a suite)

ISTR at the moment calling start more than once isn't an issue but there's potential risk of future weirdness / conflicts with other stuff.

Reply:
Added a state flag to only run it on the first execution. Was wrongly under the impression there was a new session per scenario.